nibabel/streamlines/tests/test_tractogram.py

```diff
@@ -181,6 +181,21 @@ def test_getitem(self):
             assert_arrays_equal(sdict[-1][k], v[-1])
             assert_arrays_equal(sdict[[0, -1]][k], v[[0, -1]])
 
+    def test_extend(self):
```

Comment:
Didn't see the need for them att the time. I don't think these dicts are going to be used extensively but I can add them if you want.